src/kernel/arch/x86/syscalls.zig

```diff
@@ -0,0 +1,70 @@
+const arch = @import("arch.zig");
+const testing = @import("std").testing;
+const assert = @import("std").debug.assert;
+const isr = @import("isr.zig");
+const log = @import("../../log.zig");
+
+pub const INTERRUPT: u16 = 0x80;
+pub const NUM_HANDLERS = 256;
+
+/// A syscall handler
+pub const SyscallHandler = fn (ctx: *arch.InterruptContext) void;
+/// Errors that syscall utility functions can throw
+pub const SyscallError = error {
+    SyscallExists
+};
+
+var handlers = []?SyscallHandler{null} ** NUM_HANDLERS;
+
+fn handle(ctx: *arch.InterruptContext) void {
+    // The syscall number is put in eax
+    const syscall = ctx.eax;
+    if (syscall < NUM_HANDLERS) {
+        if (handlers[syscall]) |handler| {
+            handler(ctx);
+        } else {
+            log.logWarning("Syscall {} triggered but not registered\n", syscall);
+        }
+    }
+}
+
+///
+/// Register a syscall so it  can be called by triggering interrupt 128
+///
+/// Arguments:
+///     IN syscall: u8 - The syscall number to register the handler with
+///     IN handler: SyscallHandler - The handler to register the syscall with
+///
+pub fn registerSyscall(syscall: u8, handler: SyscallHandler) !void {
+    if (handlers[syscall]) |_|
```

Comment:
Not sure if this should be an error if wanting to re-register a syscall, but i guess u'll only be registering one syscall anyways, just a though if there is a need to re-register. Also could add a bounds check for the syscall number.

Reply:
Will add a bounds check. We probably don't want to allow re-registering syscalls as then a malicious process could override some basic syscall to run its own code instead.

Reply:
someone has got there security hat on :), i guess i can do the same with the irq handlers as well, as dont want malicious code to be re-registering irq's